distributed/collections.py

```diff
@@ -145,7 +163,8 @@ def sorted(self) -> Iterator[T]:
         elements in order, from smallest to largest according to the key and insertion
         order.
         """
-        for _, _, vref in sorted(self._heap):
+        self._heap.sort()  # A sorted list maintains the heap invariant
```

Comment:
You could have a _sorted flag that is set to True after a sort() and to False after a heappush(), so that you don't need to pay O(n) to resort an already-sorted collection

Reply:
https://github.com/dask/distributed/pull/6949